### ISSUE TITLE
Very confused on how to do basic aggregations effectively

### ISSUE THREAD

**Issue by bohnman** [influxdata/ifql#145]
For reference, I am using the NOAA Weather dataset from [here](https://docs.influxdata.com/influxdb/v0.10/sample_data/data_download/).

I want to get a simple sum and count of the temperature.

My query is this:

```
curl -XPOST --data-urlencode 'q=select(db:"NOAA_water_database").range(start:0).where(exp:{"_measurement" == "h2o_temperature" AND "_field" == "degrees"}).sum().count()' localhost:8093/query
```

I get the following results:

```
h2o_temperature,location=coyote_creek degrees=494245 1439856000000000000
h2o_temperature,location=santa_monica degrees=497353 1439856000000000000
h2o_temperature,location=coyote_creek degrees=7604i 1439856000000000000
h2o_temperature,location=santa_monica degrees=7654i 1439856000000000000
```

Two problems I have:

1) The sum and count values are on different rows.  How do I know when one the sum ends and the counts start?  Why can't I have count and sum on the same row?

2) What if I want just the overall sum and the overall count regardless of location?  How do I expression this?

**Comment by nathanielc** [comment]
For the first issue this is something we are addressing currently with #143 The basic idea will be to communicate to the client the specific result boundaries.

For the second issue this is because IFQL by default does `group by *`, meaning that each series is treated separately. You can use the `group` function to change the grouping. Since you want everything to be in one large group simply call `group` without arguments.

```
select(db:"NOAA_water_database")
  .range(start:0)
  .where(exp:{"_measurement" == "h2o_temperature" AND "_field" == "degrees"})
  .group()
  .sum()
  .count()
```

**Comment by bohnman** [comment]
Regarding 1)

 That is an improvement.  However, is there any way to do some sort of projection/transform of the output so that  the sum and count are on the same row/record so it would be something like:

```
h2o_temperature,location=coyote_creek sum=12345,count=26 1439856000000000000
```
I understand that it's not the new line format, but it would be super convenient.  Otherwise I have to merge them on the client side.


Regarding 2)

I ran the exact query you specified, it returned no output

```
 curl -XPOST --data-urlencode 'q=select(db:"NOAA_water_database").range(start:0).where(exp:{"_measurement" == "h2o_temperature" AND "_field" == "degrees"}).group().sum().count()' localhost:8093/query
```

When I removed the group() call, it return the results I previously had posted.  Maybe just a bug?

**Comment by nathanielc** [comment]
We are working on ways in the client to do just that. Currently each point will only have a single value.

For 2 yeah, its probably just a bug, I'll look into it. Thanks

**Comment by nathanielc** [comment]
@bohnman There was a bug, its being fixed in #153.

For now you can do this:

Get count
```
curl -XPOST --data-urlencode 'q=from(db:"NOAA_water_database").range(start:0).filter(exp:{"_measurement" == "h2o_temperature" AND "_field" == "degrees"}).group(by:["_measurement", "_field"]).count()' localhost:8093/query
```

Get sum
```
curl -XPOST --data-urlencode 'q=from(db:"NOAA_water_database").range(start:0).filter(exp:{"_measurement" == "h2o_temperature" AND "_field" == "degrees"}).group(by:["_measurement", "_field"]).sum()' localhost:8093/query
```

Basically if the `_measurement` or `_field` tags where dropped via the aggregation then the code to convert the output to line protocol would fail.
